Assets/SEE/Utils/ThreadSafeHashSet.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace SEE.Utils
+{
+    /// <summary>
+    /// A thread-safe hash set.
+    /// </summary>
+    /// <typeparam name="T">the element type of the set</typeparam>
+    public class ThreadSafeHashSet<T> : IEnumerable
+    {
+        /// <summary>
+        /// Content of the set.
+        /// We use a dictionary with a dummy value to simulate a set.
+        /// </summary>
+        private readonly ConcurrentDictionary<T, bool> content = new();
+
+        /// <summary>
+        /// Adds an <paramref name="item"/> to the set.
+        /// </summary>
+        /// <param name="item">item to be added</param>
+        /// <returns>true if the item was added; false if it already existed</returns>
+        public bool Add(T item)
+        {
+            return content.TryAdd(item, true);
+        }
+
+        /// <summary>
+        /// Returns an enumerator that iterates through the set.
+        /// </summary>
+        /// <returns>enumerator for the set</returns>
+        public IEnumerator GetEnumerator()
+        {
+            return content.Keys.GetEnumerator();
+        }
+
+        /// <summary>
+        /// The elements of the set as an <see cref="IEnumerable{T}"/>.
+        /// </summary>
+        /// <returns>elements of the set</returns>
+        public IEnumerable<T> ToList()
```

Comment:
This method shouldn't be called `ToList()`, because it doesn't convert the elements to (or even returns) a list. I think the method can be removed completely after the above comment about the generic `IEnumerable<T>` interface is applied.

Reply:
I fixed it.